### ISSUE TITLE
Prometheus summary missing +Inf bucket when maximumExpectedValue is set

### ISSUE THREAD

**Issue by yarosla** [micrometer-metrics/micrometer#529]
The following test fails for Micrometer version 1.0.2:

```groovy
def "distribution summary limit max value"() {
    setup:
    PrometheusMeterRegistry prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT)
    def summary = DistributionSummary.builder('my.summary')
            .publishPercentileHistogram()
            .maximumExpectedValue(25)
            .register(prometheusRegistry)
    when:
    summary.record(10)
    summary.record(12)
    def scrape = prometheusRegistry.scrape()
    log.info 'scrape: {}', scrape
    then:
    scrape.contains 'my_summary_count 2.0'
    scrape.contains 'my_summary_bucket{le="9.0",} 0.0'
    scrape.contains 'my_summary_bucket{le="10.0",} 1.0'
    scrape.contains 'my_summary_bucket{le="12.0",} 2.0'
    scrape.contains 'my_summary_bucket{le="+Inf",} 2.0' // fails here
}
```

While full scrape looks like this:
```
# HELP my_summary  
# TYPE my_summary summary
my_summary_bucket{le="1.0",} 0.0
my_summary_bucket{le="2.0",} 0.0
my_summary_bucket{le="3.0",} 0.0
my_summary_bucket{le="4.0",} 0.0
my_summary_bucket{le="5.0",} 0.0
my_summary_bucket{le="6.0",} 0.0
my_summary_bucket{le="7.0",} 0.0
my_summary_bucket{le="8.0",} 0.0
my_summary_bucket{le="9.0",} 0.0
my_summary_bucket{le="10.0",} 1.0
my_summary_bucket{le="11.0",} 1.0
my_summary_bucket{le="12.0",} 2.0
my_summary_bucket{le="13.0",} 2.0
my_summary_bucket{le="14.0",} 2.0
my_summary_bucket{le="16.0",} 2.0
my_summary_bucket{le="21.0",} 2.0
my_summary_bucket{le="25.0",} 2.0
my_summary_count 2.0
my_summary_sum 22.0
my_summary_max 12.0
```

`+Inf` bucket is essential for Prometheus. `histogram_quantile()` function does not work without it ([docs](https://prometheus.io/docs/prometheus/latest/querying/functions/#histogram_quantile)):

> The highest bucket must have an upper bound of +Inf. (Otherwise, NaN is returned.)

**Comment by jkschneider** [comment]
Thanks @yarosla. Already fixed this as part of the work on #516. Hope to release 1.0.3 either tonight or tomorrow.

**Comment by jkschneider** [comment]
Fixed with 535b2aa4898f708730cc2336ed3dc8478f24360c
